src/main/scala/com/cloudera/sparkts/DateTimeIndex.scala

```diff
@@ -228,11 +228,23 @@ class IrregularDateTimeIndex(
   override def dateTimeAtLoc(loc: Int): DateTime = new DateTime(instants(loc), dateTimeZone)
 
   override def locAtDateTime(dt: DateTime): Int = {
-    java.util.Arrays.binarySearch(instants, dt.getMillis)
+    val loc = java.util.Arrays.binarySearch(instants, dt.getMillis)
+
+    if (loc < 0) {
+      return -1
```

Comment:
nit: you don't need to use "return" here.  Also, this is compact enough that it can go on one line.  Otherwise looks good.